app/models/role.rb

```diff
@@ -31,6 +31,7 @@ class Role < ActiveRecord::Base
     compare = 'not' if args.first
     where("#{compare} builtin = 0")
   }
+  scope :cloned, -> { where('cloned_from_id IS NOT NULL') }
```

Comment:
`where.not(:cloned_from_id => nil)` would be slightly better. The reason is that if scopes are chained, there is no reference to the roles table on cloned_from_id. 

```

irb(main):004:0> Role.where.not(:cloned_from_id => nil).to_sql
=> "SELECT \"roles\".* FROM \"roles\" WHERE (\"roles\".\"cloned_from_id\" IS NOT NULL)"
irb(main):005:0> Role.where('cloned_from_id IS NOT NULL').to_sql
=> "SELECT \"roles\".* FROM \"roles\" WHERE (cloned_from_id IS NOT NULL)"
```